File: C++/OrdenarPrecios.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{

    int n;

    //Leemos n mientras no sea menor a 0
    do{
        cin >> n;
    } while(n <= 0);

    vector<int> Precios(n);
    //leemos los precios

    bool error = false;
    for (int i = 0; i < n; i++)
    {
        cin >> Precios[i];
        if (Precios[i] <= 0)
        {
            cout << "Error en el precio ingresado." << endl;
            error = true;
            break;
        }
        
    }
    

    if (!error)
    {
        sort(Precios.begin(), Precios.end(), greater<int>()); //el greater nos ayuda a ordnar e mayor a menor
        for (int precio : Precios) //imprimirmos precios
        { 
            cout << precio << " ";
        }
        cout << endl;
    }
    





    return 0;
}

File: Python/OrdenarPrecios.py
def main():
    # Leer n mientras no sea menor o igual a 0
    while True:
        n = int(input())
        if n > 0:
            break

    precios = []
    error = False

    # Leer los precios
    for i in range(n):
        precio = int(input())
        if precio <= 0:
            print("Error en el precio ingresado.")
            error = True
            break
        precios.append(precio)

    # Si no hubo errores, ordenar e imprimir los precios de mayor a menor
    if not error:
        precios.sort(reverse=True)
        print(" ".join(map(str, precios)))

if __name__ == "__main__":
    main()
